### ISSUE TITLE
bug: When access token is invalid, chat hangs

### ISSUE THREAD

**Issue by dominiccooney** [sourcegraph/jetbrains#1023]
### Installation Information

RustRover 2023.3 EAP
Build #RR-233.13135.127, built on January 23, 2024
Licensed to RustRover EAP user: Dominic Cooney
Expiration date: March 23, 2024
Runtime version: 17.0.9+7-b1087.9 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.2.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 12
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  transferSettings.vscode.onlyCargoToml=true
Non-Bundled Plugins:
  com.sourcegraph.jetbrains (5.3.3914)

### Describe the bug

1. Sign in with an account
2. Invalidate the access token
3. Reload the plugin (I did an update)
4. Chat to Cody

### Expected behavior

Since the access token is invalid, I should get a message to log in.

### Additional context

This exception occurs:

```
java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: accessing Sourcegraph GraphQL API: Error: HTTP status code 401: Invalid access token.
 (https://sourcegraph.com/.api/graphql?CurrentUserCodyProEnabled)

Error: accessing Sourcegraph GraphQL API: Error: HTTP status code 401: Invalid access token.
 (https://sourcegraph.com/.api/graphql?CurrentUserCodyProEnabled)
    at /snapshot/dist/agent.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /snapshot/dist/agent.js
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at com.sourcegraph.cody.chat.ui.LLMDropdown.fetchAndUpdateModels$lambda$1(LLMDropdown.kt:49)
	at com.sourcegraph.cody.agent.CodyAgentService$Companion.withAgent$lambda$0(CodyAgentService.kt:125)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:249)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: accessing Sourcegraph GraphQL API: Error: HTTP status code 401: Invalid access token.
 (https://sourcegraph.com/.api/graphql?CurrentUserCodyProEnabled)

Error: accessing Sourcegraph GraphQL API: Error: HTTP status code 401: Invalid access token.
 (https://sourcegraph.com/.api/graphql?CurrentUserCodyProEnabled)
    at /snapshot/dist/agent.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /snapshot/dist/agent.js
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	... 1 more
```

**Comment by mkondratek** [comment]
Hello @dominiccooney  👋 thanks for the report

You can also try to regenerate your token. 

This issue should be fixed in [Sourcegraph Cody + Code Search 5.4.358](https://plugins.jetbrains.com/plugin/9682-sourcegraph-cody--code-search/edit/versions/stable/503926). Please try it 🤞

**Comment by mkondratek** [comment]
@dominiccooney , sorry for confusion - we just took a close look into this one 🕵️ 

It is very likely that the plugin update will not help you with this problem (however it fixes a few other issues so it is still worth to update 😄). 

The problem is probably related to the fact the your token has been deleted or just expired (and our communication of this fact is - let's say - for from perfect at the moment 😉 no worries - we are working in improving that 🔧 ). 

**Please, try to generate a new token.** You can do it like this:
1. Go to `Preferences | Tools | Sourcegraph & Cody`
2. Choose your account 
3. Right-click, `Edit account`
4. Clock, `Generate new token`
5. Customise (rename, etc) and `+ Generate token`
6. Copy the token value and past it in the token field in the IDE.
7.  `Save account`

I hope it helps. Let us know if you need more assistance 🙏 

(the wrapping issue: https://github.com/sourcegraph/jetbrains/issues/1035)
